sql-plugin/src/main/320+-nondb/scala/org/apache/spark/sql/delta/rapids/shims/GpuOptimisticTransaction.scala

```diff
@@ -164,8 +158,20 @@ class GpuOptimisticTransaction
     val (data, partitionSchema) = performCDCPartition(inputData)
     val outputPath = deltaLog.dataPath
 
-    val (queryExecution, output, generatedColumnConstraints, _) =
+    val (normalizedQueryExecution, output, generatedColumnConstraints, _) =
       normalizeData(deltaLog, data)
+
+    // Build a new plan with a stub GpuDeltaWrite node to work around undesired transitions between
+    // columns and rows when AQE is involved. Without this node in the plan, AdaptiveSparkPlanExec
+    // could be the root node of the plan. In that case we do not have enough context to know
+    // whether the AdaptiveSparkPlanExec should be columnar or not, since the GPU overrides do not
+    // see how the parent is using the AdaptiveSparkPlanExec outputs. By using this stub node, the
+    // parent node is present and it will be planned accordingly. We could force the AQE node to be
+    // columnar by explicitly replacing the node, but that breaks the connection between the
+    // queryExecution and the AdaptiveSparkPlanExec node that will actually execute.
```

Comment:
Do we want to explain how it avoids the AQE transition so if that stops working in the future we can understand what is happening? I don't think it will stop, but who knows.

Reply:
I updated this and the other places, along with updating the Delta Lake support docs to mention the new, custom node.  I didn't elaborate on the workaround nature of the node in the user docs since I thought that would be too much information, and IMO the node justifies itself with the ability to see stats just on the write rather than stats that include the write along with a bunch of other stuff (i.e.: the stats on the higher-level SQL operation that runs this plan as part of it).